--- drbd-8.3.2/scripts/Makefile~	2009-08-15 13:48:25.915925403 +0200
+++ drbd-8.3.2/scripts/Makefile	2009-08-15 13:51:02.752576022 +0200
@@ -104,7 +104,7 @@
 	mkdir -p $(PREFIX)/etc/xen/scripts
 	install -m 755 block-drbd $(PREFIX)/etc/xen/scripts
 	mkdir -p $(PREFIX)/etc/udev/rules.d
-ifneq ($(INST_COMPLETION),1)
+ifeq ($(INST_COMPLETION),1)
 	mkdir -p $(PREFIX)/etc/bash_completion.d
 	install -m 644 drbdadm.bash_completion $(PREFIX)/etc/bash_completion.d/$(COMPLETION)
 endif
